src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionPreviewDecisionServiceTestBase.java

```diff
@@ -944,18 +945,73 @@ public void givenWelsh_GeneratedDateIsAlreadySet_thenDoNotSetNewGeneratedDate()
+        sscsCaseData.getSscsFinalDecisionCaseData().setWriteFinalDecisionGenerateNotice("yes");
+        setHigherRateScenarioFields(sscsCaseData);
+        sscsCaseData.getSscsFinalDecisionCaseData().setWriteFinalDecisionDateOfDecision("2018-10-10");
+        sscsCaseData.getPostHearing().getCorrection().setCorrectionFinalDecisionInProgress(YesNo.YES);
+
+        service.preview(callback, DocumentType.CORRECTION_GRANTED, USER_AUTHORISATION, true, false, false);
+
+        NoticeIssuedTemplateBody payload = verifyTemplateBody(NoticeIssuedTemplateBody.ENGLISH_IMAGE, "Appellant Lastname", null, "2018-10-10", true,
+                true, false, isDescriptorFlowSupported(), true, false, documentConfiguration.getDocuments().get(LanguagePreference.ENGLISH).get(EventType.ISSUE_FINAL_DECISION));
+
+        assertEquals(LocalDate.now(), payload.getDateIssued());
+    }
+
+    @Test
+    public void givenPostHearingsIsTrueAndDocumentIsFinalDecision_thenCreateFinalDecisionDocument() {
+        setDescriptorFlowIndicator(isDescriptorFlowSupported() ? "yes" : "no", sscsCaseData);
+        sscsCaseData.getSscsFinalDecisionCaseData().setWriteFinalDecisionGenerateNotice("yes");
+        setHigherRateScenarioFields(sscsCaseData);
+        sscsCaseData.getSscsFinalDecisionCaseData().setWriteFinalDecisionDateOfDecision("2018-10-10");
+        sscsCaseData.getSscsFinalDecisionCaseData().setFinalDecisionIssuedDate(LocalDate.now());
+
+        service.preview(callback, DocumentType.FINAL_DECISION_NOTICE, USER_AUTHORISATION, true, true, true);
+
+        NoticeIssuedTemplateBody payload = verifyTemplateBody(NoticeIssuedTemplateBody.ENGLISH_IMAGE, "Appellant Lastname", null, "2018-10-10", true,
+                true, false, isDescriptorFlowSupported(), true, false, documentConfiguration.getDocuments().get(LanguagePreference.ENGLISH).get(EventType.ISSUE_FINAL_DECISION));
+
+        assertEquals(LocalDate.now(), payload.getDateIssued());
+    }
+
     protected abstract boolean isDescriptorFlowSupported();
 
     protected NoticeIssuedTemplateBody verifyTemplateBody(String image, String expectedName, String expectedAppointeeName, String dateOfDecision, boolean allowed, boolean isSetAside, boolean isDraft,
-        boolean isDescriptorFlow, boolean isGenerateFile, String templateId) {
+                                                          boolean isDescriptorFlow, boolean isGenerateFile, String templateId) {
+        return verifyTemplateBody(image, expectedName, expectedAppointeeName, dateOfDecision, allowed, isSetAside, isDraft, isDescriptorFlow, isGenerateFile, false, templateId);
+    }
+
+    protected NoticeIssuedTemplateBody verifyTemplateBody(String image, String expectedName, String expectedAppointeeName, String dateOfDecision, boolean allowed, boolean isSetAside, boolean isDraft,
+        boolean isDescriptorFlow, boolean isGenerateFile, boolean isCorrection, String templateId) {
```

Comment:
I have refactored this method in my branch to replace the boolean isDraft with a local Enum FinalDecisionType. There were a lot of usages across all the tests so much refactoring (regex replacement helped). In hindsight, an overloaded method with an extra boolean arg for isCorrection was probably the better choice. Either way, our two branches will need to be aligned.

Reply:
I will leave my branch as so then refactor yours when I take it over